pkg/ccl/importccl/import_job.go

```diff
@@ -1243,9 +1248,17 @@ func emitImportJobEvent(
 }
 
 func constructSchemaAndTableKey(
-	tableDesc *descpb.TableDescriptor, schemaIDToName map[descpb.ID]string,
+	ctx context.Context,
+	tableDesc *descpb.TableDescriptor,
+	schemaIDToName map[descpb.ID]string,
+	version clusterversion.Handle,
 ) (schemaAndTableName, error) {
 	schemaName, ok := schemaIDToName[tableDesc.GetUnexposedParentSchemaID()]
+	if !version.IsActive(ctx, clusterversion.PublicSchemasWithDescriptors) {
```

Comment:
can we move this check above the line where we check existence in the map

Reply:
Done.

> Can we cleanup the commit and PR message a little bit? The import commit mentions `restore`, and the restore commit mentions `backup` which doesn't seem to have any code change?
> 
> The PR message can just be an aggregation of the two commit messages then.

Squashed the two commit messages.